File: 0077-combinations/0077-combinations.java
import java.util.*;

class Solution {
    public List<List<Integer>> combine(int n, int k) {
        List<List<Integer>> res=new ArrayList<>();
        backtrack(1,new ArrayList<>(),res,n,k);
        return res;
    }

    private static void backtrack(int st,List<Integer> path,List<List<Integer>> res,int n,int k){
        if(path.size() == k){
            res.add(new ArrayList<>(path));
            return;
        }
        for(int i=st; i<n+1 ;i++){
            path.add(i);
            backtrack(i+1,path,res,n,k);
            path.remove(path.size()-1);
        }
    }
}

File: 0077-combinations/0077-combinations.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        def backtrack(s,p):
            if len(p)==k:
                res.append(p[:])
                return
            for i in range(s,n+1):
                p.append(i)
                backtrack(i+1,p)
                p.pop()
        res=[] 
        backtrack(1,[])
        return res